frontend/src/components/TagFilter/index.tsx

```diff
@@ -0,0 +1,220 @@
+            try {
+                const bookRes = await axios.get(`/thirdPartyBookApi/googleBooks/searchOneById/${bookId}`);
+                let bookWithNextStoryTags = bookRes.data;
+                const thisMongoBookMaybe = mongoBooks.filter((b) => b.bookId === bookId);
+                const thisMongoBook = thisMongoBookMaybe.length > 0 && thisMongoBookMaybe[0];
+                if (thisMongoBook) {
+                    bookWithNextStoryTags.nextStoryTags = thisMongoBook.nextStoryTags;
+                } else {
+                    bookWithNextStoryTags.nextStoryTags = [];
+                }
+                bookData.push(bookWithNextStoryTags);
+                setAllResults((allResults) => [...allResults, bookWithNextStoryTags]);
+            } catch (e) {
+                console.log(e);
+            }
+        }
+        if (movieData.length === 0) {
+            setHasMoreMovieResults(false);
+        }
+        if (bookData.length === 0) {
+            setHasMoreBookResults(false);
+        }
+        return [movieData, bookData];
+    }
+
+    const doNext = () => {
+        console.log('set next');
+        setQueryStartIndex(queryStartIndex + increaseIndexBy);
+    };
+
+    const tabFilter = (
+        <div>
+            <TypeSelector>
+                <Typography>
+                    Types:
+                </Typography>
+                <div>
+                    <FormControlLabel
+                        onClick={(event) => event.stopPropagation()}
+                        onFocus={(event) => event.stopPropagation()}
+                        control={
+                        <Checkbox 
+                            name="movie" 
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                selectMovie(!isMovieSelected);
+                            }} 
+                        />}
+                        label="Movies"
+                    />
+                    <FormControlLabel
+                        onClick={(event) => event.stopPropagation()}
+                        onFocus={(event) => event.stopPropagation()}
+                        control={
+                        <Checkbox
+                            name="book"
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                selectBook(!isBookSelected);
+                            }} 
+                        />}
+                        label="Books"
+                    />
+                </div>
+            </TypeSelector>
+            <div>
+                <Typography>
+                    Tags:
+                </Typography>
+                <div>
+                    {
+                        allTags.map(
+                            (tag) => (
+                                <FormControlLabel
+                                    onClick={(event) => event.stopPropagation()}
```

Comment:
I think we should give each mapped item a key (maybe key={index}?) so that we don't get those errors (React requirement)